In [7]:
import pandas as pd
import matplotlib.pylab as plt

Kode di bawah ini untuk menampilkan data yang telah diambil dan membaginya ke dalam 2 data, yaitu df untuk data mentahnya, sedangkan ext untuk data yang akan diolah

In [13]:
import pandas as pd
import numpy as np
df = pd.read_csv("\Aljut_ProyekAkhir\imdb-dataset-of-top-1000-movies-and-tv-shows\imdb_top_1000.csv")
ext = df
#df adalah data mentah, dan data ext merupakan data yang akan diproses dan dianalisis
df.head

<bound method NDFrame.head of                                            Poster_Link  \
0    https://m.media-amazon.com/images/M/MV5BMDFkYT...   
1    https://m.media-amazon.com/images/M/MV5BM2MyNj...   
2    https://m.media-amazon.com/images/M/MV5BMTMxNT...   
3    https://m.media-amazon.com/images/M/MV5BMWMwMG...   
4    https://m.media-amazon.com/images/M/MV5BMWU4N2...   
..                                                 ...   
995  https://m.media-amazon.com/images/M/MV5BNGEwMT...   
996  https://m.media-amazon.com/images/M/MV5BODk3Yj...   
997  https://m.media-amazon.com/images/M/MV5BM2U3Yz...   
998  https://m.media-amazon.com/images/M/MV5BZTBmMj...   
999  https://m.media-amazon.com/images/M/MV5BMTY5OD...   

                 Series_Title Released_Year Certificate  Runtime  \
0    The Shawshank Redemption          1994           A  142 min   
1               The Godfather          1972           A  175 min   
2             The Dark Knight          2008          UA  152 min   
3

Kode di bawah untuk mengecek nilai null untuk selanjutnya diproses

In [14]:
missing_data = df.isnull()
for column in missing_data.columns.values.tolist():
    print(column)
    print (missing_data[column].value_counts())
    print("")    

Poster_Link
Poster_Link
False    1000
Name: count, dtype: int64

Series_Title
Series_Title
False    1000
Name: count, dtype: int64

Released_Year
Released_Year
False    1000
Name: count, dtype: int64

Certificate
Certificate
False    899
True     101
Name: count, dtype: int64

Runtime
Runtime
False    1000
Name: count, dtype: int64

Genre
Genre
False    1000
Name: count, dtype: int64

IMDB_Rating
IMDB_Rating
False    1000
Name: count, dtype: int64

Overview
Overview
False    1000
Name: count, dtype: int64

Meta_score
Meta_score
False    843
True     157
Name: count, dtype: int64

Director
Director
False    1000
Name: count, dtype: int64

Star1
Star1
False    1000
Name: count, dtype: int64

Star2
Star2
False    1000
Name: count, dtype: int64

Star3
Star3
False    1000
Name: count, dtype: int64

Star4
Star4
False    1000
Name: count, dtype: int64

No_of_Votes
No_of_Votes
False    1000
Name: count, dtype: int64

Gross
Gross
False    831
True     169
Name: count, dtype: int64



Menganalisis kolom yang memiliki nilai NULL untuk menentukan metode yang optimal untuk memanipulasi datanya

In [17]:
from scipy import stats
ext.dtypes
#ext['Certificate'].unique()
#stats.mode(ext['Certificate'])

Poster_Link       object
Series_Title      object
Released_Year     object
Certificate       object
Runtime           object
Genre             object
IMDB_Rating      float64
Overview          object
Meta_score       float64
Director          object
Star1             object
Star2             object
Star3             object
Star4             object
No_of_Votes        int64
Gross             object
dtype: object

Kolom yang memiliki missing value adalah Gross, Meta-score, dan Certificate.
akan dilakukan beberapa operasi untuk mengolah data yang hilang (bernilai NaN / NULL):
1. Baris yang data Grossnya bernilai NULL akan di-drop karena Gross adalah nilai yang akan dianalisa
2. Baris yang data Certificatenya bernilai NULL akan di-drop karena Certificate akan digunakan sebagai dependensi analisis
3. Baris yang data Meta-scorenya bernilai NULL akan diganti menggunakan nilai Meta-score rata-rata

Menghapus baris yang tidak memiliki nilai Gross atau Gross==NULL (akan menghapus 169 data sehingga data setelah dihapus = 831 data)

In [18]:
# simply drop whole row with NaN in "price" column
ext.dropna(subset=["Gross"], axis=0, inplace=True)
# reset index, because we droped two rows
ext.reset_index(drop=True, inplace=True)

Memeriksa jumlah baris data saat ini untuk memastikan data yang Gross==NULL dihapus (index = 830 = 831 data)

In [19]:
ext.tail()

Poster_Link           Series_Title  \
826  https://m.media-amazon.com/images/M/MV5BYjRmY2...           Giù la testa   
827  https://m.media-amazon.com/images/M/MV5BMzAyND...         Kelly's Heroes   
828  https://m.media-amazon.com/images/M/MV5BMjAwMT...        The Jungle Book   
829  https://m.media-amazon.com/images/M/MV5BZjQyMG...     A Hard Day's Night   
830  https://m.media-amazon.com/images/M/MV5BM2U3Yz...  From Here to Eternity   

    Released_Year Certificate  Runtime                         Genre  \
826          1971          PG  157 min           Drama, War, Western   
827          1970          GP  144 min        Adventure, Comedy, War   
828          1967           U   78 min  Animation, Adventure, Family   
829          1964           U   87 min        Comedy, Music, Musical   
830          1953      Passed  118 min           Drama, Romance, War   

     IMDB_Rating                                           Overview  \
826          7.6  A low-life bandit and an I.R.A. explosives exp...   
827          7.6  A group of U.S. soldiers sneaks across enemy l...   
828          7.6  Bagheera the Panther and Baloo the Bear have a...   
829          7.6  Over two "typical" days in the life of The Bea...   
830          7.6  In Hawaii in 1941, a private is cruelly punish...   

     Meta_score             Director           Star1             Star2  \
826        77.0         Sergio Leone     Rod Steiger      James Coburn   
827        50.0      Brian G. Hutton  Clint Eastwood     Telly Savalas   
828        65.0  Wolfgang Reitherman     Phil Harris   Sebastian Cabot   
829        96.0       Richard Lester     John Lennon    Paul McCartney   
830        85.0       Fred Zinnemann  Burt Lancaster  Montgomery Clift   

               Star3             Star4  No_of_Votes        Gross  
826     Romolo Valli       Maria Monti        30144      696,690  
827      Don Rickles  Carroll O'Connor        45338    1,378,435  
828      Louis Prima  Bruce Reitherman       166409  141,843,612  
829  George Harrison       Ringo Starr        40351   13,780,024  
830     Deborah Kerr        Donna Reed        43374   30,500,000

Memeriksa nilai unik pada baris "Certificate"

In [20]:
ext['Certificate'].unique()

array(['A', 'UA', 'U', 'R', 'G', 'Passed', 'PG-13', 'PG', nan, 'Unrated',
       'GP', 'Approved', 'TV-PG', 'U/A'], dtype=object)

In [21]:
# simply drop whole row with NaN in "price" column
ext.dropna(subset=["Certificate"], axis=0, inplace=True)
# reset index, because we droped two rows
ext.reset_index(drop=True, inplace=True)

Memeriksa data yang baru saja dihapus, yaitu Certificate==NULL

In [22]:
ext.tail()

Poster_Link           Series_Title  \
780  https://m.media-amazon.com/images/M/MV5BYjRmY2...           Giù la testa   
781  https://m.media-amazon.com/images/M/MV5BMzAyND...         Kelly's Heroes   
782  https://m.media-amazon.com/images/M/MV5BMjAwMT...        The Jungle Book   
783  https://m.media-amazon.com/images/M/MV5BZjQyMG...     A Hard Day's Night   
784  https://m.media-amazon.com/images/M/MV5BM2U3Yz...  From Here to Eternity   

    Released_Year Certificate  Runtime                         Genre  \
780          1971          PG  157 min           Drama, War, Western   
781          1970          GP  144 min        Adventure, Comedy, War   
782          1967           U   78 min  Animation, Adventure, Family   
783          1964           U   87 min        Comedy, Music, Musical   
784          1953      Passed  118 min           Drama, Romance, War   

     IMDB_Rating                                           Overview  \
780          7.6  A low-life bandit and an I.R.A. explosives exp...   
781          7.6  A group of U.S. soldiers sneaks across enemy l...   
782          7.6  Bagheera the Panther and Baloo the Bear have a...   
783          7.6  Over two "typical" days in the life of The Bea...   
784          7.6  In Hawaii in 1941, a private is cruelly punish...   

     Meta_score             Director           Star1             Star2  \
780        77.0         Sergio Leone     Rod Steiger      James Coburn   
781        50.0      Brian G. Hutton  Clint Eastwood     Telly Savalas   
782        65.0  Wolfgang Reitherman     Phil Harris   Sebastian Cabot   
783        96.0       Richard Lester     John Lennon    Paul McCartney   
784        85.0       Fred Zinnemann  Burt Lancaster  Montgomery Clift   

               Star3             Star4  No_of_Votes        Gross  
780     Romolo Valli       Maria Monti        30144      696,690  
781      Don Rickles  Carroll O'Connor        45338    1,378,435  
782      Louis Prima  Bruce Reitherman       166409  141,843,612  
783  George Harrison       Ringo Starr        40351   13,780,024  
784     Deborah Kerr        Donna Reed        43374   30,500,000

Menghitung nilai rata-rata "Meta_score"

In [23]:
avg_meta_score = ext['Meta_score'].astype("float").mean(axis=0)
print("Rata-rata meta-score adalah :", avg_meta_score)

Rata-rata meta-score adalah : 77.15826330532212


Mereplace data yang memiliki Meta_score==NULL dengan nilai rata-rata Meta_score film yang memiliki Certificate dan Gross

In [24]:
ext["Meta_score"].replace(np.nan, avg_meta_score, inplace=True)

In [29]:
#Menghapus kolom overview
ext = ext.drop('Overview', axis=1)

KeyError: "['Overview'] not found in axis"

In [31]:
#Menghapus kolom poster_link
ext = ext.drop('Poster_Link', axis=1)

In [32]:
ext.head()

Series_Title Released_Year Certificate  Runtime  \
0  The Shawshank Redemption          1994           A  142 min   
1             The Godfather          1972           A  175 min   
2           The Dark Knight          2008          UA  152 min   
3    The Godfather: Part II          1974           A  202 min   
4              12 Angry Men          1957           U   96 min   

                  Genre  IMDB_Rating  Meta_score              Director  \
0                 Drama          9.3        80.0        Frank Darabont   
1          Crime, Drama          9.2       100.0  Francis Ford Coppola   
2  Action, Crime, Drama          9.0        84.0     Christopher Nolan   
3          Crime, Drama          9.0        90.0  Francis Ford Coppola   
4          Crime, Drama          9.0        96.0          Sidney Lumet   

            Star1           Star2          Star3           Star4  No_of_Votes  \
0     Tim Robbins  Morgan Freeman     Bob Gunton  William Sadler      2343110   
1   Marlon Brando       Al Pacino     James Caan    Diane Keaton      1620367   
2  Christian Bale    Heath Ledger  Aaron Eckhart   Michael Caine      2303232   
3       Al Pacino  Robert De Niro  Robert Duvall    Diane Keaton      1129952   
4     Henry Fonda     Lee J. Cobb  Martin Balsam    John Fiedler       689845   

         Gross  
0   28,341,469  
1  134,966,411  
2  534,858,444  
3   57,300,000  
4    4,360,000

In [37]:
#Tren Peringkat IMDB
imdb_stats = ext['IMDB_Rating'].describe()
print("Statistik Peringkat IMDB:")
print(imdb_stats)

Statistik Peringkat IMDB:
count    785.000000
mean       7.948408
std        0.287540
min        7.600000
25%        7.700000
50%        7.900000
75%        8.100000
max        9.300000
Name: IMDB_Rating, dtype: float64


In [39]:
#Hubungan antara IMDB Rating dan Meta Score
correlation = ext['IMDB_Rating'].corr(ext['Meta_score'])
print("Korelasi antara IMDB Rating dan Meta Score:")
print(correlation)

Korelasi antara IMDB Rating dan Meta Score:
0.27321892543483217


In [40]:
#Distribusi Genre Film
genre_counts = ext['Genre'].value_counts()
print("Distribusi Genre Film:")
print(genre_counts)

Distribusi Genre Film:
Genre
Drama                        64
Drama, Romance               29
Comedy, Drama                26
Comedy, Drama, Romance       25
Biography, Drama, History    23
                             ..
Drama, History                1
Drama, Film-Noir, Mystery     1
Adventure, Thriller           1
Animation, Action, Sci-Fi     1
Adventure, Comedy, War        1
Name: count, Length: 178, dtype: int64


In [43]:
#Pengaruh Durasi Terhadap Peringkat
runtime_rating_corr = df['Runtime'].corr(df['IMDB_Rating'])
print("Korelasi antara Durasi dan Peringkat IMDB:")
print(runtime_rating_corr)


TypeError: unsupported operand type(s) for /: 'str' and 'int'

In [44]:
actors = ['Star1', 'Star2', 'Star3', 'Star4']
for actor in actors:
    actor_rating_mean = df.groupby(actor)['IMDB_Rating'].mean()
    print(f"Rata-rata peringkat IMDB berdasarkan {actor}:")
    print(actor_rating_mean)

Rata-rata peringkat IMDB berdasarkan Star1:
Star1
Aamir Khan              8.216667
Aaron Taylor-Johnson    7.600000
Abhay Deol              8.000000
Adam Driver             7.900000
Adrian Molina           8.400000
                          ...   
Zach Galifianakis       7.700000
Zain Al Rafeea          8.400000
Zbigniew Zamachowski    7.600000
Zooey Deschanel         7.700000
Éric Toledano           8.500000
Name: IMDB_Rating, Length: 518, dtype: float64
Rata-rata peringkat IMDB berdasarkan Star2:
Star2
Adil Hussain            7.8
Adèle Exarchopoulos     7.7
Adèle Haenel            8.1
Aidan Gillen            7.9
Akira Terao             8.2
                       ... 
Zach Galifianakis       7.7
Zachary Quinto          7.8
Zack Gottsagen          7.6
Zbigniew Zamachowski    7.9
Zoe Saldana             7.7
Name: IMDB_Rating, Length: 661, dtype: float64
Rata-rata peringkat IMDB berdasarkan Star3:
Star3
Aamir Khan         8.4
Aaron Eckhart      9.0
Abhay Deol         8.1
Adel Bencherif  

In [63]:
# 6. Tren Pendapatan

director_gross_mean = df.groupby('Director')['Gross'].mean()

print("Rata-rata pendapatan berdasarkan direktur:")
print(director_gross_mean)

Rata-rata pendapatan berdasarkan direktur:
Director
Aamir Khan               1223869.0
Abdellatif Kechiche      2199675.0
Abhishek Kapoor          1122527.0
Adam McKay              70259870.0
Aditya Dhar              4186168.0
                          ...     
Yimou Zhang             28156540.0
Yoshiaki Kawajiri         151086.0
Yôjirô Takita            1498210.0
Zack Snyder            159062369.0
Zoya Akhtar              4337509.5
Name: Gross, Length: 444, dtype: float64
